compiler/z/codegen/OMRTreeEvaluator.cpp

```diff
@@ -6121,17 +6111,37 @@ aloadHelper(TR::Node * node, TR::CodeGenerator * cg, TR::MemoryReference * tempM
                   generateRXInstruction(cg, TR::InstOpCode::LLGF, node, tempReg, tempMR);
                else
                   {
-                  // TODO (GuardedStorage): Do we need the useCompressedPointers check here? All aloads should have been lowered by now.
+                  auto loadMnemonic = TR::InstOpCode::BAD;
+
                   if (cg->isConcurrentScavengeEnabled() &&
-                      !comp->useCompressedPointers() &&
                       node->getOpCodeValue() == TR::aloadi &&
                       tempReg->containsCollectedReference())
                      {
-                     generateRXYInstruction(cg, TR::InstOpCode::LGG, node, tempReg, tempMR);
+                     if (comp->useCompressedPointers())
+                        {
+                        if (TR::TransformUtil::fieldShouldBeCompressed(node, comp))
```

Comment:
@vijaysun-omr FYI this is why its impotent that this API never returns false negatives.